src/PerformanceSensitiveAnalyzers/CSharp/CallSiteImplicitAllocationAnalyzer.cs

```diff
@@ -79,6 +79,11 @@ private static void CheckParam(InvocationExpressionSyntax invocationExpression,
             var arguments = invocationExpression.ArgumentList.Arguments;
             if (arguments.Count == methodInfo.Parameters.Length - 1)
             {
+                // Up to net45 the System.Array.Empty<T> singleton didn't existed so an empty params array was still causing some memory allocation.
+                if (semanticModel.Compilation.GetSpecialType(SpecialType.System_Array).GetMembers("Empty").Length == 0)
```

Comment:
I have tried to extract this to the `AnalyzeNode` method but this was actually causing a few more un-needed access. The best thing would be to have this check done once at the base-class level but I wasn't sure this was ok for you to update everything.

Reply:
I'm OK with this for now since it's already gated on the attribute.